clusterloader/framework/util.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+// CreatePods creates pods in user defined namspaces with user configurable tuning sets
+func CreatePods(f *framework.Framework, appName string, ns string, spec v1.PodSpec, maxCount int, tuning *TuningSetType) {
+	for i := 0; i < maxCount; i++ {
+		framework.Logf("%v/%v : Creating container", i+1, maxCount)
+		labels := map[string]string{"purpose": "test"}
+		_, err := f.ClientSet.Core().Pods(ns).Create(&v1.Pod{
+			ObjectMeta: v1.ObjectMeta{
+				Name:      fmt.Sprintf(appName+"-pod-%v", i),
+				Namespace: ns,
+				Labels:    labels,
+			},
+			Spec: spec,
+		})
+		framework.ExpectNoError(err)
+		if tuning != nil {
```

Comment:
Do you mean incorporate the create & wait in this single function? The idea behind the rate limit delay is to push pods as fast as the apiserver can handle them. On the other hand the stepping size does wait until the pods are running.